### PR TITLE
gh-112169: Documented getaddrinfo/getnameinfo default loop executor usage and implications.

### PR DESCRIPTION
**Pull request by alex-kowalczyk**
Proposing additional note for `asyncio.getaddrinfo`/`asyncio.getnameinfo`, to document they employ the loops's default executor which might manifest in "DNS timeouts" when the executor is saturated.
See also: Guido's remark at https://github.com/python/cpython/issues/112169#issuecomment-1814927606

* Issue: gh-112169

----
:books: Documentation preview :books:: https://cpython-previews--112191.org.readthedocs.build/

### CODE REVIEW

File: Doc/library/asyncio-eventloop.rst [hunk 1]
```diff
@@ -1147,6 +1147,13 @@ DNS
 
    Asynchronous version of :meth:`socket.getnameinfo`.
 
+.. note::
+   Both *getaddrinfo* and *getnameinfo* internally utilize their synchronous
+   versions through the loop's default thread pool executor.
+   When this executor is saturated, these methods may experience delays,
+   which can manifest as "DNS timeouts" in higher-level networking libraries.
+   To mitigate this, consider using a custom executor for other user tasks.
```
Comment (gvanrossum) [inline Doc/library/asyncio-eventloop.rst]:
```suggestion
   To mitigate this, consider using a custom executor for other user tasks,
   or setting a default executor with a larger number of workers.
```

**Review by gvanrossum — APPROVED** [review]
Approved, with one tiny addition of my own: you could possibly just configure a bigger thread pool. I'll wait merging for a bit in case you object to this suggestion.

**Comment by gvanrossum** [comment]
Thinking about this some more, I'm curious what exactly you mean by "DNS timeouts". Is this referring to some library calling e.g. `getaddrinfo()` and reporting an error if it doesn't respond within a specific time interval? It's slightly incorrect to flag that as a "DNS timeout" (since the problem is on the application side, not on the DNS side -- at least from the network's POV). I don't see how a saturated executor could lead to *actual* DNS timeouts, where DNS itself doesn't respond timely to requests. Maybe the documentation note should be more explicit about what kind of errors can happen as a result? It is indeed the case that calls to `await loop.getaddrinfo(...)` might not return in a timely fashion.

File: Doc/library/asyncio-eventloop.rst [hunk 2]
```diff
@@ -1147,6 +1147,14 @@ DNS
 
    Asynchronous version of :meth:`socket.getnameinfo`.
 
+.. note::
+   Both *getaddrinfo* and *getnameinfo* internally utilize their synchronous
+   versions through the loop's default thread pool executor.
+   When this executor is saturated, these methods may experience delays,
+   which can manifest as "DNS timeouts" in higher-level networking libraries.
```
Comment (willingc) [inline Doc/library/asyncio-eventloop.rst]:
```suggestion
   which higher-level networking libraries may report as increased timeouts.
```

**Review by willingc — COMMENTED** [review]
Addressing @gvanrossum's comment, my suggestion would be to remove the reference to "DNS timeouts" since higher level networking libraries could treat in different ways.

**Review by kumaraditya303 — APPROVED** [review]

**Pull request merged**